zpay32/invoice_test.go

```diff
@@ -515,10 +529,66 @@ func TestDecodeEncode(t *testing.T) {
 				return i
 			},
 		},
+		{
+			// Decode a mainnet invoice while expecting active net to be testnet
+			encodedInvoice: "lnbc241pveeq09pp5qqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypqdqqnp4q0n326hr8v9zprg8gsvezcch06gfaqqhde2aj730yg0durunfhv66jd3m5klcwhq68vdsmx2rjgxeay5v0tkt2v5sjaky4eqahe4fx3k9sqavvce3capfuwv8rvjng57jrtfajn5dkpqv8yelsewtljwmmycq62k443",
+			valid:          false,
+			decodedInvoice: func() *Invoice {
+				return &Invoice{
+					Net:         &chaincfg.TestNet3Params,
+					MilliSat:    &testMillisat24BTC,
+					Timestamp:   time.Unix(1503429093, 0),
+					PaymentHash: &testPaymentHash,
+					Destination: testPubKey,
+					Description: &testEmptyString,
+				}
+			},
+			skipEncoding: true, // Skip encoding since we were given the wrong net
+		},
+		{
+			// Decode a litecoin testnet invoice
+			encodedInvoice: "lntltc241pvjluezpp5qqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypqhp58yjmdan79s6qqdhdzgynm4zwqd5d7xmw5fk98klysy043l2ahrqsnp4q0n326hr8v9zprg8gsvezcch06gfaqqhde2aj730yg0durunfhv66m2eq2fx9uctzkmj30meaghyskkgsd6geap5qg9j2ae444z24a4p8xg3a6g73p8l7d689vtrlgzj0wyx2h6atq8dfty7wmkt4frx9g9sp730h5a",
+			valid:          true,
+			decodedInvoice: func() *Invoice {
+				return &Invoice{
+					// TODO(sangaman): create an interface for chaincfg.params
+					Net:             &ltcTestNetParams,
+					MilliSat:        &testMillisat24BTC,
+					Timestamp:       time.Unix(1496314658, 0),
+					PaymentHash:     &testPaymentHash,
+					DescriptionHash: &testDescriptionHash,
+					Destination:     testPubKey,
+				}
+			},
+		},
+		{
+			// Decode a litecoin mainnet invoice
+			encodedInvoice: "lnltc241pvjluezpp5qqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypqhp58yjmdan79s6qqdhdzgynm4zwqd5d7xmw5fk98klysy043l2ahrqsnp4q0n326hr8v9zprg8gsvezcch06gfaqqhde2aj730yg0durunfhv66859t2d55efrxdlgqg9hdqskfstdmyssdw4fjc8qdl522ct885pqk7acn2aczh0jeht0xhuhnkmm3h0qsrxedlwm9x86787zzn4qwwwcpjkl3t2",
+			valid:          true,
+			decodedInvoice: func() *Invoice {
+				return &Invoice{
+					Net:             &ltcMainNetParams,
+					MilliSat:        &testMillisat24BTC,
+					Timestamp:       time.Unix(1496314658, 0),
+					PaymentHash:     &testPaymentHash,
+					DescriptionHash: &testDescriptionHash,
+					Destination:     testPubKey,
+				}
+			},
+		},
 	}
 
 	for i, test := range tests {
-		invoice, err := Decode(test.encodedInvoice)
+		var decodedInvoice *Invoice
+		var net *chaincfg.Params
```

Comment:
You can just set this `= mainnet`, then drop the `else` below.

Reply:
Good point.